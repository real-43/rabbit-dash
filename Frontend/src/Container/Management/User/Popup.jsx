import { useState } from "react"

function popup() {

    const [newName, setNewName] = useState(props.currentName)
    const [newPassword, setNewPassword] = useState(props.currentPassword)
    const [role, setRole] = useState(props.currentRole)
    const [isActive, setActive] = useState(false)
    const [isOpen, setIsOpen] = useState(props.isOpen)

    // To show/unshow pass in edit popup
    function displayOption() {
        setActive(!isActive);
        var x = document.getElementById("myInput");

        if (x.type === "password") {
            x.type = "text";
        } else {
            x.type = "password";
        }
    }
    
    return (props.isOpen) ? (
      <div>
        <Modal show={true} onHide={(e)=>{props.onClose}}>
          <Modal.Header>
            <Modal.Title>Edit User<i onClick={(e) => {props.onClose}} style={{cursor:"pointer", marginLeft:"330px"}} className='fa fa-times'/></Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <Form>
              <Form.Group className="mb-3" controlId="exampleForm.ControlInput1">
                <Form.Label>Name</Form.Label>
                <Form.Control
                  type="text"
                  placeholder={props.changeUserName}
                  value={newName}
                  onChange={(e) => {props.nameChange}}
                  autoFocus
                />
              </Form.Group>
              <Form.Group
                className="mb-3"
                controlId="exampleForm.ControlTextarea1"
              >
                <Form.Label>Password</Form.Label>
                <InputGroup className="mb-3">
                  
                  <FormControl
                    aria-label="Example text with button addon"
                    aria-describedby="basic-addon1"
                    id='myInput'
                    type="password"
                    placeholder="password"
                    value={newPassword}
                    onChange={(e) => props.passChange}
                  />
                  <Button variant="outline-secondary" id="button-addon1" onClick={(e) => displayOption(this)}>
                    <i class={isActive ? "fa fa-eye" : "fa fa-eye-slash"} id="togglePassword"/>
                  </Button>
                </InputGroup>
              </Form.Group>
              <Form.Label>Role</Form.Label>
              <Form.Select aria-label={role} defaultValue={role} onChange={(e) => setRole(e.target.value)}>
                <option className="d-none" value="">{role}</option>
                <option value="">Set No role</option>
                {roles.filter(obj => checkDomainRole(obj)).map((role,i) => {return (
                  <option value={role.name} key={i}>{role.name}</option>
                )})}
              </Form.Select>
            </Form>
          </Modal.Body>
          <Modal.Footer>
            <Button variant="secondary" onClick={(e) => {props.onClose}}>
              Close
            </Button>
            <Button variant="primary" onClick={(e) =>{handleChange(changeUser)}}>
              Save Changes
            </Button>
          </Modal.Footer>
        </Modal>
      </div>
    ) : ""
  }