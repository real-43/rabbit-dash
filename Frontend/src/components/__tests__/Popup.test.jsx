import { React } from "react";
import Popup from "../../Container/Management/User/EditPopup.jsx"
import { mount } from 'enzyme';

it('renders content when modal is open', () => {
    // mount renders to the dom (real or mocked)
    const wrapper = mount(<Popup />);
    wrapper.find('button').simulate('click');
  
    // element text
    expect(wrapper.find(Modal).text()).toBe('modal content');
});

it('render Popup component correctly', () => {  
    const Popup = shallow(<Popup />);
    expect(Popup).toMatchSnapshot();
});