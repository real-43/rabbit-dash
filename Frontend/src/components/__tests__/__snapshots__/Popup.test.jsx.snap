// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Popup component correctly 1`] = `ShallowWrapper {}`;
